run-tests.php

```diff
@@ -1209,6 +1268,339 @@ function run_all_tests($test_files, $env, $redir_tested = null)
 	}
 }
 
+function run_all_tests_parallel($test_files, $env, $redir_tested) {
+	global $workers, $test_idx, $test_cnt, $test_results, $failed_tests_file, $result_tests_file, $PHP_FAILED_TESTS;
+
+	// The PHP binary running run-tests.php, and run-tests.php itself
+	// This PHP executable is *not* necessarily the same as the tested version
+	$thisPHP = PHP_BINARY;
+	$thisScript = __FILE__;
+
+	$workerProcs = [];
+	$workerStdins = [];
+	$workerStdouts = [];
+	$workerStderrs = [];
+
+	echo "====⚡️===========================================================⚡️====\n";
```

Comment:
I don't think the emoji here is a good idea. it will cause problems in Windows/CMD.

Reply:
I wouldn't recommend `proc_open` on Windows anyway, because STDIO are file handles there and block.

Reply:
Quick testing on Windows 10 shows that in `cmd`, the emoji don't cause problems, they just show up as empty rectangles. That's fine, really. Maybe it can serve as an incentive to support Unicode terminal output.

And @kelunik is right about `proc_open` blocking… it effectively behaves as if there's only one worker as-is. :/

Reply:
**empty rectangles** IS a problem, isn't it? If you insist on these emoji, I suggest you add something like `--no-ansi`, and turn it on on windows.

Reply:
It's just cosmetic, so why bother? And actually, it's just a font support problem. Copy and paste them into another window and it shows up fine. I guess PHP does use Unicode after all.

Reply:
I began modifying the code so it can also communicate over sockets, but this is a little more painful than I expected. If I get that working, it should solve the Windows problem.

Reply:
…oh, I was using the wrong socket API >.>